test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/HtmlBlockTest/HandlesUnbalancedTripleDashHTMLComments.stree.txt

```diff
@@ -1,52 +1,56 @@
-                    SyntaxKind.Text;[that];
-                    SyntaxKind.Whitespace;[ ];
-                    SyntaxKind.Text;[shouldn];
-                    SyntaxKind.SingleQuote;['];
-                    SyntaxKind.Text;[t];
-                    SyntaxKind.Whitespace;[ ];
-                    SyntaxKind.Text;[break];
-                    SyntaxKind.Whitespace;[ ];
-                    SyntaxKind.Text;[razor];
-                    SyntaxKind.Whitespace;[ ];
-                    SyntaxKind.Text;[-];
-                Markup span - Gen<Markup> - [-->] - SpanEditHandler;Accepts:None - (62:1,58) - Tokens:2
-                    SyntaxKind.DoubleHyphen;[--];
-                    SyntaxKind.CloseAngle;[>];
-            Markup span - Gen<Markup> - [LF] - SpanEditHandler;Accepts:None - (65:1,61) - Tokens:1
-                SyntaxKind.NewLine;[LF];
-        Code span - Gen<Stmt> - [] - SpanEditHandler;Accepts:Any - (67:2,0) - Tokens:1
-            SyntaxKind.Unknown;[];
-        MetaCode span - Gen<None> - [}] - SpanEditHandler;Accepts:None - (67:2,0) - Tokens:1
-            SyntaxKind.RightBrace;[}];
-    Markup span - Gen<Markup> - [] - SpanEditHandler;Accepts:Any - (68:2,1) - Tokens:1
-        SyntaxKind.Unknown;[];
+RazorDocument - [0..68) - FullWidth: 68 - [@{LF    <!-- Hello, I'm a comment that shouldn't break razor --->LF}]
+    MarkupBlock - [0..68) - FullWidth: 68
+        MarkupTextLiteral - [0..0) - FullWidth: 0 - Gen<Markup> - SpanEditHandler;Accepts:Any
+            Marker;[];
+        CSharpCodeBlock - [0..68) - FullWidth: 68
+            CSharpStatement - [0..68) - FullWidth: 68
+                CSharpTransition - [0..1) - FullWidth: 1 - Gen<None> - SpanEditHandler;Accepts:None
+                    Transition;[@];
+                CSharpStatementBody - [1..68) - FullWidth: 67
+                    RazorMetaCode - [1..2) - FullWidth: 1 - Gen<None> - SpanEditHandler;Accepts:None
+                        LeftBrace;[{];
+                    CSharpCodeBlock - [2..67) - FullWidth: 65
+                        CSharpStatementLiteral - [2..4) - FullWidth: 2 - Gen<Stmt> - AutoCompleteEditHandler;Accepts:Any,AutoComplete:[<null>];AtEOL
+                            NewLine;[LF];
+                        MarkupBlock - [4..67) - FullWidth: 63
+                            MarkupTextLiteral - [4..8) - FullWidth: 4 - Gen<Markup> - SpanEditHandler;Accepts:Any
+                                Whitespace;[    ];
+                            MarkupCommentBlock - [8..65) - FullWidth: 57
+                                MarkupTextLiteral - [8..12) - FullWidth: 4 - Gen<Markup> - SpanEditHandler;Accepts:None
+                                    OpenAngle;[<];
+                                    Bang;[!];
+                                    DoubleHyphen;[--];
+                                MarkupTextLiteral - [12..62) - FullWidth: 50 - Gen<Markup> - SpanEditHandler;Accepts:Whitespace
+                                    Whitespace;[ ];
```

Comment:
Might be easier to read if you swap the order of these and add some space
```
[ ];Whitespace;
[Hello,];  Text
[ ];  Whitespace;
[I];  Text;
```

Reply:
I thought I'll keep the same ordering as before. I guess it might help to display the content of all `*Literal` nodes right next to it? Would that be helpful?

Reply:
Yup, read my mind 😉